homeassistant/components/bsblan/config_flow.py

```diff
@@ -41,6 +46,55 @@ async def async_step_user(
         self.username = user_input.get(CONF_USERNAME)
         self.password = user_input.get(CONF_PASSWORD)
 
+        return await self._validate_and_create()
+
+    async def async_step_zeroconf(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle Zeroconf discovery."""
+
+        self.host = str(
+            getattr(discovery_info, "ip_address", None)
+            or (discovery_info.ip_addresses[0] if discovery_info.ip_addresses else None)
+            or discovery_info.hostname
+        )
+        self.port = discovery_info.port or DEFAULT_PORT
+
+        # Store discovery info for later use
+        self._discovered_device = {
+            CONF_HOST: self.host,
+            CONF_PORT: self.port,
+        }
+
```

Comment:
iirc you already need to abort here, otherwise the user will get the pop up for the same device again and again

Reply:
Thanks! I could not get the MAC address I use for unique ID. I need to make a request that this information is in the zeroconf protocol I guess. So I check now the ip address and port. I hope this is good enough?